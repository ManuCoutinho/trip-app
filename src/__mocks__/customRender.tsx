import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
