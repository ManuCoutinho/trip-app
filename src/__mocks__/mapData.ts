import { MapProps } from 'components/Map/types'

export default {
  places: [
    {
      id: 'first-place',
      name: 'Rio de Janeiro',
      slug: 'rio-de-janeiro',
      location: {
        latitude: -22.925453,
        longitude: -43.277763
      }
    },
    {
      id: 'second-place',
      name: 'Lagos',
      slug: 'lagos',
      location: {
        latitude: 6.460649,
        longitude: 3.333309
      }
    }
  ]
} as MapProps
