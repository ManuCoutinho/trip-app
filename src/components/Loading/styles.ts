import styled, { css, keyframes } from 'styled-components'

const dotAnimation = keyframes`
  0% {
  transform: scale(1) rotate(0);
  }
  50% {
  transform: scale(2) rotate(-180deg);
  }
  100% {
  transform: scale(1) rotate(-360deg) translateY(-10px);
  }
`
const dotUpAnimation = keyframes`
  0% {
  transform: scale(1) rotate(0);
  }
  50% {
  transform: scale(2) rotate(180deg);
  }
  100% {
  transform: scale(1) rotate(360deg) translateY(10px);
  }
`

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100vw;
    height: 100vh;
    background: ${theme.colors.background};
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 5rem;
  `}
`
export const Dot = styled.div<{ color: string; up?: boolean }>`
  ${({ color, up }) => css`
    width: 15px;
    height: 15px;
    background: ${color};
    border-radius: 50%;

    animation: ${up ? dotUpAnimation : dotAnimation} 2s ease-in 0.5s infinite
      forwards;
  `}
`
