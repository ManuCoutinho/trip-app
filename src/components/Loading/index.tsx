import { useTheme } from 'styled-components'
import * as Styled from './styles'

const Loading: React.FC = () => {
  const { colors } = useTheme()
  return (
    <Styled.Wrapper
      data-testid='loading'
      role='status'
      aria-live='polite'
      aria-atomic={true}
      aria-labelledby='loading'
    >
      <Styled.Dot color={colors.highlight} up />
      <Styled.Dot color={colors.secondary} />
      <Styled.Dot color={colors.highlight} up />
      <Styled.Dot color={colors.secondary} />
      <Styled.Dot color={colors.highlight} up />
    </Styled.Wrapper>
  )
}

export default Loading
