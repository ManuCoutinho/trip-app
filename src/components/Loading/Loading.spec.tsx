import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import Loading from '.'

describe('<Loading />', () => {
  it('should render the Loading wrapper correctly', () => {
    render(<Loading />)
    expect(screen.getByRole('status')).toBeInTheDocument()
  })
  it('should match to snapshot', () => {
    const { container } = render(<Loading />)

    expect(container).toMatchSnapshot()
  })
})
