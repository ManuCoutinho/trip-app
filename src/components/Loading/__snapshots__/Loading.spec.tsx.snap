// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match to snapshot 1`] = `
<div>
  <section
    aria-atomic="true"
    aria-labelledby="loading"
    aria-live="polite"
    class="sc-bcXHqe ewlqML"
    data-testid="loading"
    role="status"
  >
    <div
      class="sc-gswNZR iMqyrN"
      color="#fc4949"
    />
    <div
      class="sc-gswNZR dtutif"
      color="#6d6d6d"
    />
    <div
      class="sc-gswNZR iMqyrN"
      color="#fc4949"
    />
    <div
      class="sc-gswNZR dtutif"
      color="#6d6d6d"
    />
    <div
      class="sc-gswNZR iMqyrN"
      color="#fc4949"
    />
  </section>
</div>
`;
